### ISSUE TITLE
Log related crash

### ISSUE THREAD

**Issue by eriksundin** [Tealium/tealium-ios#28]
Hi,

We are seeing this crash happening in 5.1.1 (upgraded from 4.x version).
I'm wondering about the -[TEALLogger logDev:] there. We've set our configuration in the Tealium Dashboard to 'prod' level logging - but that sounds like a dev logging call.

Any ideas on the crash and dev log?

Thanks,
Erik

```
SIGABRT ABORT 0x0000000190f6f014

Crashed: com.tealium.operations.queue.schibsted.blocket-ios.prod
0  libsystem_kernel.dylib         0x190f6f014 __pthread_kill + 8
1  libsystem_pthread.dylib        0x191037450 pthread_kill + 112
2  libsystem_c.dylib              0x190ee3400 abort + 140
3  libsystem_malloc.dylib         0x190fb3a5c _nano_vet_and_size_of_live + 330
4  libsystem_malloc.dylib         0x190fb4f18 nano_realloc + 376
5  libsystem_malloc.dylib         0x190fa7240 malloc_zone_realloc + 180
6  Foundation                     0x1929f2628 -[NSString quotedStringRepresentation] + 732
7  Foundation                     0x1929d244c -[NSString _stringRepresentation] + 376
8  CoreFoundation                 0x191ec8eec -[NSDictionary descriptionWithLocale:indent:] + 996
9  Foundation                     0x1929acaf4 _NSDescriptionWithLocaleFunc + 76
10 CoreFoundation                 0x191f6223c __CFStringAppendFormatCore + 7996
11 CoreFoundation                 0x191f602c8 _CFStringCreateWithFormatAndArgumentsAux2 + 244
12 Foundation                     0x1929ac968 +[NSString stringWithFormat:] + 68
13 TealiumIOS                     0x1018af738 -[TEALDispatch description] (TEALDispatch.m:126)
14 Foundation                     0x1929acb14 _NSDescriptionWithLocaleFunc + 108
15 CoreFoundation                 0x191f6223c __CFStringAppendFormatCore + 7996
16 CoreFoundation                 0x191f602c8 _CFStringCreateWithFormatAndArgumentsAux2 + 244
17 TealiumIOS                     0x1018b6160 -[TEALLogger logDev:] (TEALLogger.m:146)
18 TealiumIOS                     0x1018a2bb4 -[Tealium logDispatch:status:error:] (Tealium.m:816)
19 TealiumIOS                     0x10189f5b0 __24-[Tealium feedbackBlock]_block_invoke (Tealium.m:223)
20 TealiumIOS                     0x1018a5a64 __60-[Tealium dispatchManager:requestsDispatch:completionBlock:]_block_invoke (Tealium.m:1337)
21 libdispatch.dylib              0x190e2a1fc _dispatch_call_block_and_release + 24
22 libdispatch.dylib              0x190e2a1bc _dispatch_client_callout + 16
23 libdispatch.dylib              0x190e383dc _dispatch_queue_serial_drain + 928
24 libdispatch.dylib              0x190e2d9a4 _dispatch_queue_invoke + 652
25 libdispatch.dylib              0x190e388d8 _dispatch_queue_override_invoke + 360
26 libdispatch.dylib              0x190e3a34c _dispatch_root_queue_drain + 572
27 libdispatch.dylib              0x190e3a0ac _dispatch_worker_thread3 + 124
28 libsystem_pthread.dylib        0x1910332a0 _pthread_wqthread + 1288
29 libsystem_pthread.dylib        0x191032d8c start_wqthread + 4
```

**Comment by eriksundin** [comment]
Another related crash we're getting. Seems to go through the same code path.
```
Crashed: com.blocket.blocket.tealium-remote-command.processing
0  libsystem_kernel.dylib         0x1842db014 __pthread_kill + 8
1  libsystem_pthread.dylib        0x1843a3450 pthread_kill + 112
2  libsystem_c.dylib              0x18424f400 abort + 140
3  libsystem_malloc.dylib         0x18431fa5c _nano_vet_and_size_of_live + 330
4  libsystem_malloc.dylib         0x184320f18 nano_realloc + 376
5  libsystem_malloc.dylib         0x184313240 malloc_zone_realloc + 180
6  Foundation                     0x185d5e628 -[NSString quotedStringRepresentation] + 732
7  Foundation                     0x185d3e44c -[NSString _stringRepresentation] + 376
8  CoreFoundation                 0x185234f0c -[NSDictionary descriptionWithLocale:indent:] + 1028
9  Foundation                     0x185d18af4 _NSDescriptionWithLocaleFunc + 76
10 CoreFoundation                 0x1852ce23c __CFStringAppendFormatCore + 7996
11 CoreFoundation                 0x1852cc2c8 _CFStringCreateWithFormatAndArgumentsAux2 + 244
12 Foundation                     0x185d18968 +[NSString stringWithFormat:] + 68
13 TealiumIOS                     0x10199567c -[TEALRemoteCommandResponse description] (TEALRemoteCommandResponse.m:17)
14 Foundation                     0x185d18b14 _NSDescriptionWithLocaleFunc + 108
15 CoreFoundation                 0x1852ce23c __CFStringAppendFormatCore + 7996
16 CoreFoundation                 0x1852cc2c8 _CFStringCreateWithFormatAndArgumentsAux2 + 244
17 TealiumIOS                     0x1019b2160 -[TEALLogger logDev:] (TEALLogger.m:146)
18 TealiumIOS                     0x101997e44 __72-[Tealium(TagManagement) tagDispatchServiceShouldPermitRequest:webView:]_block_invoke (Tealium+TagManagement.m:327)
19 TealiumIOS                     0x1019b9f94 __77-[TEALRemoteCommandManager triggerCommand:response:responseBlock:completion:]_block_invoke (TEALRemoteCommandManager.m:447)
20 libdispatch.dylib              0x1841961fc _dispatch_call_block_and_release + 24
21 libdispatch.dylib              0x1841961bc _dispatch_client_callout + 16
22 libdispatch.dylib              0x1841a2f94 _dispatch_continuation_pop + 576
23 libdispatch.dylib              0x1841a2c58 _dispatch_async_redirect_invoke + 628
24 libdispatch.dylib              0x1841a634c _dispatch_root_queue_drain + 572
25 libdispatch.dylib              0x1841a60ac _dispatch_worker_thread3 + 124
26 libsystem_pthread.dylib        0x18439f2a0 _pthread_wqthread + 1288
27 libsystem_pthread.dylib        0x18439ed8c start_wqthread + 4
```

**Comment by mertid** [comment]
Please try our newest release for a fix (5.1.3)

**Comment by michaeleiselsc** [comment]
@mertid just curious because i'm having this problem for an unrelated app, was it memory corruption due to a C-level mistake?
